src/channel.ts

```diff
@@ -173,26 +176,56 @@ export function hasScale(channel: Channel) {
   return !contains([DETAIL, TEXT, ORDER], channel);
 }
 
-// Position does not work with ordinal (lookup) scale and sequential (which is only for color)
-const POSITION_SCALE_TYPE_INDEX = toSet(without(SCALE_TYPES, ['ordinal', 'sequential'] as ScaleType[]));
-
 export function supportScaleType(channel: Channel, scaleType: ScaleType): boolean {
+  // Position does not work with ordinal (lookup) scale and sequential (which is only for color)
+  const POSITION_SCALE_TYPE_INDEX = toSet(without(SCALE_TYPES, ['ordinal', 'sequential'] as ScaleType[]));
```

Comment:
This move is bad, this means you call this `toSet` and `without` every time you call `supportScaleType`.